src/js/common/schemaform/save-in-progress/FormStartControls.jsx

```diff
@@ -23,7 +23,11 @@ class FormStartControls extends React.Component {
   }
 
   handleLoadPrefill = () => {
-    if (this.props.prefillAvailable) {
+    if (this.props.beforeStartForm) {
+      this.props.beforeStartForm().then((hasValidITF) => {
+        if (hasValidITF) this.props.fetchInProgressForm(this.props.formId, this.props.migrations, true, this.props.prefillTransformer);
```

Comment:
I think instead of checking an argument in the `then` callback here it would be better to expect someone using `beforeStartForm` to reject if they don't want to start the form and resolve if they do.

Reply:
I'll second @jbalboni's approach.

Reply:
I know we have to get this out the door quickly, though, so if you write a new ticket for this, I think this PR is good to go.

(I hate operating this way, by the by. :disappointed: )

Reply:
Thanks @jbalboni @cvalarida, updated.